runelite-client/src/main/java/net/runelite/client/util/OSXUtil.java

```diff
@@ -36,14 +38,42 @@
 public class OSXUtil
 {
 	/**
-	 * Enables the osx native fullscreen if running on a mac.
+	 * Enables the osx native fullscreen if running on a Mac.
 	 *
 	 * @param gui The gui to enable the fullscreen on.
 	 */
 	public static void tryEnableFullscreen(JFrame gui)
 	{
 		if (OSType.getOSType() == OSType.MacOS)
 		{
+			FullScreenUtilities.addFullScreenListenerTo(gui, new FullScreenListener()
```

Comment:
you can use a `FullScreenAdapter` to avoid having to implement the whole interface